### PR TITLE
[INLONG-4962][DataProxy] Unify the  MQ zone sink

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Prepare a Pull Request
- Fixes https://github.com/apache/inlong/issues/4962

### Motivation&&Modifications

implement AbstractClusterZoneProducer and AbstractSinkZoneContext.
Implement the AbstractZoneProducer.
implement the AbstractZoneSink and AbstractZoneWork.

### CODE REVIEW

**Comment by liangyepianzhou** [comment]
@dockerzhang Could you please help review this PR wehn you have time? THX.

**Comment by dockerzhang** [comment]
@liangyepianzhou please fix the checkstyles error.

**Comment by liangyepianzhou** [comment]
> @liangyepianzhou please fix the checkstyles error.

Fixed.

**Comment by liangyepianzhou** [comment]
All tests have passed, please help review this PR, THX.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mqzone/ZoneWorkerCalculator.java [hunk 1]
```diff
@@ -15,11 +15,8 @@
...
 
-package org.apache.inlong.dataproxy.sink.pulsar;
+package org.apache.inlong.dataproxy.sink.mqzone;
 
-public interface CreatePulsarClientCallBack {
-
-    void handleCreateClientSuccess(String url);
-
-    void handleCreateClientException(String url);
+public interface ZoneWorkerCalculator {
```
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mqzone/ZoneWorkerCalculator.java:20]: How about `ZoneWorkerFactoryCalculator`?

File: inlong-dataproxy/conf/dataproxy-tube.conf [hunk 2]
```diff
@@ -122,37 +122,37 @@ agent1.channels.ch-msg10.fsyncPerTransaction = false
 agent1.channels.ch-msg10.fsyncInterval = 5
 
 agent1.sinks.meta-sink-more1.channel = ch-msg1
-agent1.sinks.meta-sink-more1.type = org.apache.inlong.dataproxy.sink.TubeSink
+agent1.sinks.meta-sink-more1.type = org.apache.inlong.dataproxy.sink.mqzone.impl.kafkazone.KafkaZoneSink
```
Comment (dockerzhang) [inline inlong-dataproxy/conf/dataproxy-tube.conf:125]: here should be TubeZoneSink？

Comment (liangyepianzhou) [reply inlong-dataproxy/conf/dataproxy-tube.conf:125]: Yes, I will correct it right away.

**Review by pocozh — COMMENTED** [review]
Please don't directly delete `PulsarSink` and `TubeSink`.  You can find that the implementations of `PulsarSink` and `PulsarZoneSInk` are incompatible, including some functions and interfaces. These should be added into new sink.

**Comment by pocozh** [comment]
What's more, please add necessary UT, thanks.

**Comment by liangyepianzhou** [comment]
> What's more, please add necessary UT, thanks.

The https://github.com/apache/inlong/issues/5321 is adding the missing zone sink test.

**Review by luchunliang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by pocozh — APPROVED** [review]

**Pull request merged**